src/ecmult_impl.h

```diff
@@ -640,12 +640,18 @@ static void secp256k1_ecmult(const secp256k1_ecmult_context *ctx, secp256k1_gej
 }
 
 static size_t secp256k1_strauss_scratch_size(size_t n_points) {
+    size_t size = 0;
+    size += ROUND_TO_ALIGN(n_points * sizeof(secp256k1_gej));
+    size += ROUND_TO_ALIGN(n_points * sizeof(secp256k1_scalar));
+    size += ROUND_TO_ALIGN(n_points * ECMULT_TABLE_SIZE(WINDOW_A) * sizeof(secp256k1_gej));
+    size += ROUND_TO_ALIGN(n_points * ECMULT_TABLE_SIZE(WINDOW_A) * sizeof(secp256k1_fe));
 #ifdef USE_ENDOMORPHISM
-    static const size_t point_size = (2 * sizeof(secp256k1_ge) + sizeof(secp256k1_gej) + sizeof(secp256k1_fe)) * ECMULT_TABLE_SIZE(WINDOW_A) + sizeof(struct secp256k1_strauss_point_state) + sizeof(secp256k1_gej) + sizeof(secp256k1_scalar);
+    size += ROUND_TO_ALIGN(n_points * 2 * ECMULT_TABLE_SIZE(WINDOW_A) * sizeof(secp256k1_ge));
 #else
-    static const size_t point_size = (sizeof(secp256k1_ge) + sizeof(secp256k1_gej) + sizeof(secp256k1_fe)) * ECMULT_TABLE_SIZE(WINDOW_A) + sizeof(struct secp256k1_strauss_point_state) + sizeof(secp256k1_gej) + sizeof(secp256k1_scalar);
+    size += ROUND_TO_ALIGN(n_points * ECMULT_TABLE_SIZE(WINDOW_A) * sizeof(secp256k1_ge));
 #endif
-    return n_points*point_size;
+    size += ROUND_TO_ALIGN(n_points * sizeof(struct secp256k1_strauss_point_state));
+    return size;
```

Comment:
Hm, I'm somewhat unsure about this. It seems like a layer violation to care about the alignment here.

Reply:
If we want to have a function that returns the required scratch space given a number of points (and we should) then we need to add the padding somewhere. While we can assume the worst case padding somewhere else I would prefer to have the `*_scratch_space` function return exact results. This makes it much easier to think about and also helps testing because now we can just check that what is allocated actually matches what we computed with `*_scratch_space` (see https://github.com/bitcoin-core/secp256k1/pull/638/commits/24553bfa1c34bcbe4820a783d66649b0a18affb7#diff-4655d106bf03045a3a50beefc800db21R2996). Do you have an alternative in mind?

Reply:
Added function `alloc_size` to scratch space